src/dmtcp_dlsym.cpp

```diff
@@ -575,15 +572,14 @@ dmtcp_dlsym(void *handle, const char *symbol)
                                                         NULL, &tags,
                                                         &default_symbol_index);
   print_debug_messages(tags, default_symbol_index, symbol);
-  dmtcp_enable_ckpt();
   return result;
 }
 
 EXTERNC void *
 dmtcp_dlvsym(void *handle, char *symbol, const char *version)
 {
   // Acquire checkpoint lock.
-  dmtcp_disable_ckpt();
+  dmtcp::WrapperLock wrapperExecutionLock;
```

Comment:
Please add a second comment line above:
`// C++ releases the lock at the end of the function, since this is a local var.`

Students and others not familiar with C++ would find the comment about "Acquire" with no "Release" to be confusing.  We either need the second line, or delete the comment about an Acquire with no Release.  Personally, I prefer more comments.  :-)